pygfx/objects/_base.py

```diff
@@ -452,19 +452,35 @@ def _wgpu_get_pick_info(self, pick_value):
         # In most cases the material handles this.
         return self.material._wgpu_get_pick_info(pick_value)
 
-    def look_at(self, target) -> None:
-        up = self.up
-        new_z = target - self.world.position
+    def look_at(self, target, *, up=None) -> None:
+        if up is not None:
+            self.up = up
```

Comment:
The user is not expecing `object.look_at` to have a persistent effect on `object.up`.

The kwarg `up` provided to `object.look_at` applies only for the scope of the method.

Reply:
I adopted it from our current implementation, but I'm happy to remove it if it is not meant to be this way :)

https://github.com/pygfx/pygfx/blob/974ea7ec854e7b0655de1913c949f3dc3110dbf2/pygfx/objects/_base.py#L471-L481

Reply:
I think I added it in the camera/controller refactor. The idea was to optionally also *set* the up vector. I thought I only added it for the `camera.show_x` methods (where it makes more sense). Apparently it also leaked into `look_at`.

My proposal would be to remove the `up` arg from `look_at`, and keep it for the `camera.show_x` methods (they'd simply set the prop instead of passing it to `look_at`).

Reply:
Sorry about that both, I missed it in almar's earlier PR then.

Reply:
So we drop the `self.up = up` assignment here and instead move it to each `camera.show_<thing>` instead?